src/Features/Core/Portable/Diagnostics/AbstractHostDiagnosticUpdateSource.cs

```diff
@@ -1,119 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Immutable;
-using System.Linq;
-using Microsoft.CodeAnalysis.Shared.Collections;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Diagnostics;
-
-/// <summary>
-/// Diagnostic update source for reporting workspace host specific diagnostics,
-/// which may not be related to any given project/document in the solution.
-/// For example, these include diagnostics generated for exceptions from third party analyzers.
-/// </summary>
-internal abstract class AbstractHostDiagnosticUpdateSource
-{
-    private ImmutableDictionary<DiagnosticAnalyzer, ImmutableHashSet<DiagnosticData>> _analyzerHostDiagnosticsMap =
-        ImmutableDictionary<DiagnosticAnalyzer, ImmutableHashSet<DiagnosticData>>.Empty;
-
-    public abstract Workspace Workspace { get; }
-
-    public event EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>? DiagnosticsUpdated;
```

Comment:
this was the member i was removing first.  it was only referenced from tests, so it culd be safely removed.  this then became a therad that removed more and more.